src/custom/hooks/useRefetchFee.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from 'react'
+import { useAddFee, useClearFee } from '../state/fee/hooks'
+import { registerOnWindow } from '../utils/misc'
+import { FeeQuoteParams, getFeeQuote } from '../utils/operator'
+
+/**
+ * @returns callback that fetches a new quote and update the state
+ */
+export function useRefetchFeeCallback() {
+  const addFee = useAddFee()
+  const clearFee = useClearFee()
+  registerOnWindow({ addFee })
+
+  return useCallback(
+    async ({ sellToken, buyToken, amount, kind, chainId }: FeeQuoteParams) => {
+      // Get a new quote
+      const fee = await getFeeQuote({ chainId, sellToken, buyToken, amount, kind }).catch(err => {
+        console.error(new Error(err))
```

Comment:
Do you need to wrap the error?

Reply:
Ouch, missed some comments. No, I will address in another PR. 
I don't even remember doing that!

Reply:
https://github.com/gnosis/gp-swap-ui/pull/482